lib/teleterm/clusters/cluster_servers.go

```diff
@@ -68,3 +72,84 @@ func (c *Cluster) GetServers(ctx context.Context) ([]Server, error) {
 
 	return results, nil
 }
+
+func (c *Cluster) GetServers(ctx context.Context, r *api.GetServersRequest) (*GetServersResponse, error) {
+	var (
+		clusterServers []types.Server
+		resp           *types.ListResourcesResponse
+		authClient     auth.ClientI
+		proxyClient    *client.ProxyClient
+		sortBy         types.SortBy
+		err            error
+	)
+
+	err = addMetadataToRetryableError(ctx, func() error {
+		proxyClient, err = c.clusterClient.ConnectToProxy(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		defer proxyClient.Close()
+		return nil
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	defer authClient.Close()
+
+	authClient, err = proxyClient.ConnectToCluster(ctx, c.clusterClient.SiteName)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	sortParam := r.SortBy
+	if sortParam != "" {
+		vals := strings.Split(sortParam, ":")
+		if vals[0] != "" {
+			sortBy.Field = vals[0]
+			if len(vals) > 1 && vals[1] == "desc" {
+				sortBy.IsDesc = true
+			}
+		}
+	}
+
+	resp, err = authClient.ListResources(ctx, proto.ListResourcesRequest{
+		Namespace:           defaults.Namespace,
+		ResourceType:        types.KindNode,
+		Limit:               r.Limit,
+		SortBy:              sortBy,
+		PredicateExpression: r.Query,
+		SearchKeywords:      client.ParseSearchKeywords(r.Search, ' '),
+		UseSearchAsRoles:    r.SearchAsRoles == "yes",
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	clusterServers, err = types.ResourcesWithLabels(resp.Resources).AsServers()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	results := []Server{}
+	for _, server := range clusterServers {
+		results = append(results, Server{
+			URI:    c.URI.AppendServer(server.GetName()),
+			Server: server,
+		})
+	}
+
+	return &GetServersResponse{
+		Servers:    results,
+		StartKey:   resp.NextKey,
+		TotalCount: resp.TotalCount,
+	}, nil
+}
+
+type GetServersResponse struct {
+	// Resources is a list of resource.
+	Servers []Server
+	// StartKey is the next key to use as a starting point.
+	StartKey string
```

Comment:
Why name it `StartKey` then if it's supposed to be the next key? ;f

Reply:
Following convention [from web](https://github.com/gravitational/teleport/blob/6d475e12b2edc2c11f49d00251d1266a5d540349/lib/web/apiserver.go#L2099). I agree its weird. We are receiving the "next key to start at". The backend is saying "this is the next key if you want to continue pagination" and the frontend uses it as the 'start key' for the next page.